tests/agentzh.t

```diff
@@ -113,7 +113,70 @@ print(cjson.encode(data))
 
 
 
-=== TEST 9: multiple calls to lua_cjson_new (1/2)
+=== TEST 9: array_mt on empty tables
+--- lua
+local cjson = require "cjson"
+local data = {}
+setmetatable(data, cjson.array_mt)
+print(cjson.encode(data))
+--- out
+[]
+
+
+
+=== TEST 10: array_mt on non-empty tables
+--- lua
+local cjson = require "cjson"
+local data = { "foo", "bar" }
+setmetatable(data, cjson.array_mt)
+print(cjson.encode(data))
+--- out
+["foo","bar"]
+
+
+
+=== TEST 11: array_mt on non-empty tables with holes
+--- lua
+local cjson = require "cjson"
+local data = {}
+data[1] = "foo"
+data[2] = "bar"
+data[4] = "last"
+data[9] = "none"
+setmetatable(data, cjson.array_mt)
+print(cjson.encode(data))
+--- out
+["foo","bar",null,"last"]
```

Comment:
I assumed this was the desired behavior if we wish to force the encoding of the array part.